packages/flutter/test/material/stepper_horizontal_title_test.dart

```diff
@@ -0,0 +1,97 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+
+import '../../lib/material.dart';
+
+void main() {
+  testWidgets('Stepper horizontal title [HorizontalTitlePosition.bottom] and line [HorizontalLinePosition.top]', (WidgetTester tester) async {
+    int index = 0;
+
+    await tester.pumpWidget(
+      MaterialApp(
+        home: Material(
+          child: Stepper(
+            horizontalTitlePosition: HorizontalTitlePosition.bottom,
+            horizontalLinePosition: HorizontalLinePosition.top,
+            onStepTapped: (int i) {
+              index = i;
+            },
+            steps: const <Step>[
+              Step(
+                title: Text('Step 1'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+              Step(
+                title: Text('Step 2'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+              Step(
+                title: Text('Step 3'),
+                content: SizedBox(
+                  width: 100.0,
+                  height: 100.0,
+                ),
+              ),
+            ],
+          ),
+        ),
+      ),
+    );
+    await tester.tap(find.text('Step 2'));
+    expect(index, 1);
```

Comment:
It will be important to inspect the underlying widgets and make sure everything was composed as anticipated. Also, these tests should be moved to `test/material/stepper_test.dart` where, incidentally, there are many other examples of tests that closely examine how Stepper widgets of varying configurations were built internally.